### PR TITLE
[Issue 5869][deploy] Bump AWS terraform deployment to Terraform 0.13 and AWS provider 3.3.0

### LINKED ISSUE #5869: [terraform] Error while using terraform init command in pulsar/deployment/terraform-ansible/aws using Terraform v0.12.18
I am trying to run the terraform `init command` to set-up my AWS cluster so I can deploy a Pulsar cluster, and I am getting the following error:

```
>terraform init
Initializing the backend...
Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "random" (hashicorp/random) 2.2.1...
Provider "registry.terraform.io/-/aws" v1.5.0 is not compatible with Terraform 0.12.18.
Provider version 2.7.0 is the earliest compatible version. Select it with
the following version constraint:
	version = "~> 2.7"
Terraform checked all of the plugin versions matching the given constraint:
    1.5
Consult the documentation for this provider for more information on
compatibility between provider and Terraform versions.
Error: incompatible provider version
```

I have installed the terraform on my Mac using `brew install terraform` command and the output of `terraform -version` is `Terraform v0.12.18`

### PR DESCRIPTION
**Pull request by KoenR3**
Fixes #5869

### Motivation

The Terraform code in the deployment example is deprecated with newer Terraform versions. Newever versions such as 0.13 also no longer offer support for the AWS provider that is added as a requirement.

### Modifications

- Removed string interpolation from all variables and resource names that do not need string interpolation (deprecated in Terraform)
- Defined tags as a block with "=" as the old version is no longer supported
- Use of the general splat operator that is present since 0.12 to call  aws_instance.proxy[*].id in aws_elb
- Add versions.tf as a way to keep track of versions for providers and terraform as this is the new standard in Terraform

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: ( no 
  - The default values of configurations:  no
  - The wire protocol:  no
  - The rest endpoints:  no
  - The admin cli options: no
  - Anything that affects deployment: yes

### Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable 
  - If a feature is not applicable for documentation, explain why? There is no new feature

### CODE REVIEW

**Review by bdjidi — APPROVED** [review]
Worked on my flow. Actually a small update of map => map(string) also silences all warnings on Terraform 0.14.

File: deployment/terraform-ansible/aws/variables.tf [hunk 1]
```diff
@@ -62,7 +62,7 @@ variable "num_proxy_nodes" {
 }
 
 variable "instance_types" {
-  type = "map"
+  type = map
```
Comment (coute) [inline deployment/terraform-ansible/aws/variables.tf:65]: type must be map(string)

File: deployment/terraform-ansible/aws/versions.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+      version = "3.3.0"
```
Comment (coute) [inline deployment/terraform-ansible/aws/versions.tf:5]: Don't lock used version. It could be better if version is ">= 3.3.0"

**Review by coute — CHANGES_REQUESTED** [review]
I worked on my own to fix terraform issues and i made same fixes in the code.
I just added 2 comments in my review.
With my requested changes, 
`terraform plan` then `terraform apply` commands worked fine with : 
- Terraform v1.0.0 on linux_amd64
- provider registry.terraform.io/hashicorp/aws v3.46.0
- provider registry.terraform.io/hashicorp/random v3.1.0

File: deployment/terraform-ansible/aws/variables.tf (hunk 1 above)
Comment (KoenR3) [reply deployment/terraform-ansible/aws/variables.tf]: I have changed this to map(string)

File: deployment/terraform-ansible/aws/versions.tf (hunk 2 above)
Comment (KoenR3) [reply deployment/terraform-ansible/aws/versions.tf]: Version has been "unlocked" in new commit

**Review by coute — APPROVED** [review]

**Review by coute — APPROVED** [review]

**Review by dave2wave — APPROVED** [review]
This will bring the AWS terraform deployment nearly up to date. There are additional fixes that will be needed in a new PR to make the sensible playbooks function. I've been doing a bunch of OpenMessaging Benchmark deployments.

**Pull request merged**